src/asciidoctorWebViewConverter.ts

```diff
@@ -69,9 +69,15 @@ function getCspForResource (webviewResourceProvider: WebviewResourceProvider, se
       return '<meta http-equiv="Content-Security-Policy" content="">'
 
     case AsciidocPreviewSecurityLevel.Strict:
-    default:
-      return `<meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src 'self' ${rule} https: data:; media-src 'self' ${rule} https: data:; script-src 'nonce-${nonce}' '${highlightjsInlineScriptHash}'; style-src 'self' ${rule} 'unsafe-inline' https: data:; font-src 'self' ${rule} https: data:;">`
-  }
+      // Initialize the cspRule variable with the default CSP rule
+      let cspRule = `default-src 'none'; img-src 'self' ${rule} https: data:; media-src 'self' ${rule} https: data:; script-src 'nonce-${nonce}' '${highlightjsInlineScriptHash}'; style-src 'self' ${rule} 'unsafe-inline' https: data:; font-src 'self' ${rule} https: data:;`;
+      // If asciidoc.extensions.enableKroki is true, add the kroki-server-url attribute to the CSP rule
+      if (asciidoc.extensions.enableKroki) {
```

Comment:
Unless I'm missing something `asciidoc` does not exist in this context?
I think we need to add `enableKroki` and `krokiServerUrl` to the `AsciidocPreviewConfiguration`, then pass the configuration to the `getCspForResource` method

Reply:
I also think we should add this logic in `AsciidocPreviewSecurityLevel.AllowInsecureLocalContent`

Reply:
@Mogztter, why do we need to add this logic to `AsciidocPreviewSecurityLevel.AllowInsecureLocalContent` when it will now be the default?

Reply:
This is already resolved when localhost is allowed:
```
case AsciidocPreviewSecurityLevel.AllowInsecureLocalContent:
    return `<meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src 'self' ${rule} https: data: http://localhost:* http://127.0.0.1:*; media-src 'self' ${rule} https: data: http://localhost:* http://127.0.0.1:*; script-src 'nonce-${nonce}' '${highlightjsInlineScriptHash}'; style-src 'self' ${rule} 'unsafe-inline' https: data: http://localhost:* http://127.0.0.1:*; font-src 'self' ${rule} https: data: http://localhost:* http://127.0.0.1:*;">`
```

Reply:
https://github.com/Mogztter/asciidoctor-kroki#using-your-own-kroki Notes that "Note that either the http:// or https:// prefix is required (the default Docker image only uses HTTP)." - so I am not sure about adding the logic to .AllowInsecureLocalContent

Reply:
> Unless I'm missing something `asciidoc` does not exist in this context? I think we need to add `enableKroki` and `krokiServerUrl` to the `AsciidocPreviewConfiguration`, then pass the configuration to the `getCspForResource` method

It exists in the Asciidoc parser -   https://github.com/asciidoctor/asciidoctor-vscode/blob/823bb2262a0072a3999ea2f6c15d8794ed92a3df/src/asciidocParser.ts#L318

Reply:
If `enableKroki` exists only at the parser level, should we not just add the default to the parser file, something like this - 
``` private async registerAsciidoctorExtensions (registry, cspRule: string) {
  const enableKroki = vscode.workspace.getConfiguration('asciidoc.extensions', null).get('enableKroki')
  if (enableKroki) {
    const kroki = require('asciidoctor-kroki')
    kroki.register(registry)
    cspRule += `; img-src 'self' ${kroki-server-url} data:;`
  }
  registry.preprocessor(this.prependExtension)
  await this.registerExtensionsInWorkspace(registry, cspRule)
}
```

Reply:
Or this problem is much simpler than it seems. Maybe all I need to do is change the security level once enableKroki is equal to true? - 
```
const enableKroki = vscode.workspace.getConfiguration('asciidoc.extensions', null).get('enableKroki')
if (enableKroki) {
  securityLevel = AsciidocPreviewSecurityLevel.AllowInsecureLocalContent;
  const kroki = require('asciidoctor-kroki')
  kroki.register(registry)
}
registry.preprocessor(this.prependExtension)
await this.registerExtensionsInWorkspace(registry)
}
```

Reply:
> why do we need to add this logic to AsciidocPreviewSecurityLevel.AllowInsecureLocalContent when it will now be the default?

Because the Kroki server URL should always be allowed no matter which security level is selected. The Kroki server URL can be any URL:

- `https://kroki.io`
- `http://localhost:8000`
- `https://example.org/kroki`
- `http://kroki.internal:1234` 


> If enableKroki should only change the security level when kroki-server-url has been changed, we can declare a variable - const krokiServerUrl = previewAttributes['kroki-server-url'] and then change the if statement to include a check if kroki-server-url has been set?

I think it's easier to always allow the `kroki-server-url`.

> Instead of automatically adding a rule when enableKroki is true, maybe all I need to do is change the security level once enableKroki is equal is true? Is that allowed? (It's logically the same thing happening if the default for the kroki.io extension is sending information and receives diagrams back from https//kroki.io, the user is deciding to enable the graphs being rendered elsewhere than the VSCode space) -

I don't think we should change the security level but when `enableKroki` is true we should allow the `kroki-server-url` not matter which security level is selected.

Does it make sense?

Reply:
Okay, to make it easier to always allow the kroki-server-url, I would refactor the code under the `getCspForResource` function to be structured as cspRules, and then add this if statement to append the cspRule no matter which security level is selected:
```
// Check if the enableKroki configuration is set to true
  if (vscode.workspace.getConfiguration('asciidoc.extensions', null).get('enableKroki')) {
    // Append the kroki-server-url to the CSP rule
    cspRule += `; img-src 'self' ${kroki-server-url};`
  }
```

Reply:
Sounds good 👍🏻

Reply:
@Mogztter Plz review when you have time new commit on this branch for these edits